OpenRA.Mods.Common/Widgets/ExpSliderWidget.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System;
+using System.Drawing;
+using OpenRA.Graphics;
+using OpenRA.Widgets;
+
+namespace OpenRA.Mods.Common.Widgets
+{
+	public class ExpSliderWidget : SliderWidget
+	{
+		public ExpSliderWidget() { }
+		public ExpSliderWidget(ExpSliderWidget other) : base(other) { }
+
+		float ExpFromLinear(float x)
+		{
+			return (float)(x == MinimumValue ? MinimumValue : 1.0e-3 * Math.Exp(6.908 * x)).Clamp(MinimumValue, MaximumValue);
```

Comment:
Why not just drop the ternary operator if you're going to clamp it anyway?

Reply:
because e^x does not produce 0 at x=0 (the result of above function is 0.001), meaning you are not able to mute sound. 
Clamp was actually meant for approximation before, because it turned out slider value is derived from mouse position relative to slider widget BEFORE clamping, meaning it produced negative values. But still that same e^x produces values slightly higher than 1.0 on x=1.0 (1.0024 for the above function)

Reply:
My point stays, the ternary should go. If you want it to round to zero at the bottom, then make it so.

Reply:
So how do I round it to minimum at the bottom without ternary?

Reply:
youre completely right and im an idiot, again. Just change the equals to a less than or equals.

Reply:
Hmm... Yeah, I was assuming the span of 0..1 for Min/Max. Will think of something about this

Reply:
This line is a bit of a monster.  Can you please reformat it to:
```csharp
if (x == MinimumValue)
    return MinimumValue;

return 1.0e-3 * Math.Exp(6.908 * x)).Clamp(MinimumValue, MaximumValue);
```

It might also be a good idea to pull those constants out to fields so that they can be overridden from yaml.